tox.ini

```diff
@@ -1,15 +1,14 @@
 [tox]
 #envlist = python3.7, python3.8, python3.9, python3.10, python3.11
-envlist = py311, py310, py39, py38, py37
+envlist = py312, py311, py310, py39
 isolated_build = True
 
 [gh-actions]
 python:
+  3.12: py312
   3.11: py311
   3.10: py310
   3.9: py39
```

Comment:
Earlier you specify `CIBW_PROJECT_REQUIRES_PYTHON: ">=3.10"`, should we also remove py39 and just require Python 3.10?

This is what most packages are doing, and recommended in [SPEC 0](https://scientific-python.org/specs/spec-0000/).